### PR TITLE
Bring Coverage of build.py to 100%

### PR DESCRIPTION
**Pull request by hynek**
Half-assed coverage as now just makes for flaky CI.

Let's take it to 100% either by writing the tests or giving up and calling it a `# pragma: no cover`

### CODE REVIEW

**Comment by adiroiban** [comment]
I am not a fan of using `# pragma: no cover` to fake a 100% coverage.

I think we can leave it as it is.

next time someone needs to modify that line, it will be a good opportunity to write a missing test.

We can merge this even if we don't have 100% coverage.

Less than 100% coverage is bad, but for an open source project is nothing to be ashamed of.
Contributors are invited to help complete the coverage by writing missing tests.
So a coverage less than 100% for an open source project means - Help is wanted :)

For a commercial project less than 100% means - Warning, some corners were cut!

**Comment by exarkun** [comment]
I'm confused.  The PR is a draft but also a review is requested.  And the description talks about "pragma: nocover" but the diff adds no such lines.

**Comment by adiroiban** [comment]
This is due to GitHub CODEOWNER automation.

if you create a PR and don't mark it as draft from the start, it will ask for a review.

> [hynek](https://github.com/hynek) requested a review from https://github.com/orgs/twisted/teams/twisted-contributors as a [code owner](https://github.com/twisted/towncrier/blob/ccedc04ebdfffa456b23e1d748e2b2edbfddc4aa/.github/CODEOWNERS#L2) [1 hour ago](https://github.com/twisted/towncrier/pull/416#event-7307494718)

Here, the PR was first created as "ready" and then was converted to "draft".



Github automation is simple, and the previously requested review was no revoked.

**Comment by hynek** [comment]
I have honestly never found a way to make a PR a draft right away and then go and look for "make a draft" forever which is tiny text. I wish it were a prominent checkbox somewhere.

---

As for no covers: I mean cases that can't be realistically or usefully tested. Having 100% coverage is nothing to be proud of (I consider it table stakes and have removed the badge from my projects), it's just a useful state that you've tested everything you want to be tested and don't have to rely on flaky Codecov mathematics.

---

All that said, it's probably best to do this piece-meal so let's merge this for build.py

**Comment by hynek** [comment]
lol of course codecov is hanging

**Comment by hynek** [comment]
(I've restarted the build, but Codecov patch is hibernating it seems :()

**Review by twm — APPROVED** [review]
The new test looks reasonable (I did not verify that it gets us to 100% coverage, hopefully codecov will wake up and do that). Thank you!

**Comment by hynek** [comment]
```
Name                              Stmts   Miss Branch BrPart  Cover   Missing
-----------------------------------------------------------------------------
src/towncrier/_git.py                15      2      8      3    78%   13, 18, 23->exit
src/towncrier/_project.py            38      6     16      5    76%   45, 51, 56, 71, 76-78
src/towncrier/_settings/load.py      71      1     34      2    97%   28->31, 36
src/towncrier/_writer.py             23      0     18      1    98%   36->39
src/towncrier/check.py               67      2     26      1    97%   127-128
src/towncrier/create.py              44      0     20      1    98%   119->121
-----------------------------------------------------------------------------
TOTAL                               553     11    230     13    97%
```

**Comment by adiroiban** [comment]
There is an error with the new toml and coverage doesn't work and is not reported as part of the CI.

I wish the CI would fail in this case.

```
    Error running `['/opt/hostedtoolcache/Python/3.10.6/x64/bin/python', '-m', 'coverage', 'xml', '-i']`: returncode=1, output=b"Can't read 'pyproject.toml' without TOML support. Install with [toml] extra\n"
    - Ignored: [Errno 2] No such file or directory: '/home/runner/work/towncrier/towncrier/coverage.xml'
Error: No coverage report found
```

**Comment by hynek** [comment]
Thanks Adi!

File: .github/workflows/ci.yml
Comment (adiroiban) [inline .github/workflows/ci.yml:99]: I wish pip had something like `pip install --only-deps` 

But maybe, we can just do `pip install .[dev]` and run the tests without tox.

**Review by adiroiban — APPROVED** [review]
All good. Thanks. Let's merge. Project coverage can be ignored.

The patch coverage is important.

**Pull request merged**
